bemenu: add module

omernaveedxyz opened the pull request:
### Description

Add bemenu module https://github.com/Cloudef/bemenu with ability to configure options through `BEMENU_OPTS` environment variable.

Configuration of bemenu globally is done through setting this environment variable as opposed to using a config file as mentioned [here](https://github.com/Cloudef/bemenu/issues/15) and [here](https://github.com/Cloudef/bemenu/issues/158).

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

#### Maintainer CC

<!--
If you are updating a module, please @ people who are in its `meta.maintainers` list.
If in doubt, check `git blame` for whoever last touched something.
-->

ambroisie commented on tests/modules/programs/bemenu/basic-configuration.nix:
```diff
@@ -0,0 +1,25 @@
+
+  nmt.script = ''
+    assertFileExists home-path/etc/profile.d/hm-session-vars.sh
+    assertFileContent home-path/etc/profile.d/hm-session-vars.sh \
+      ${./basic-configuration.sh}
```
This will lead to needless churn if the surrounding content of `hm-session-vars.sh` changes. You should probably use `assertFileContains` instead.

omernaveedxyz replied:
I've updated the tests to use `assertFileContains` and `assertFileNotRegex` instead. Let me know if there is a better way to handle this.

omernaveedxyz commented:
I've updated the `settings` options to allow boolean values for settings such as `ignorecase`. I've also updated the test cases and example to reflect this.

rycee commented:
Thanks! Merged to master now 🙂
